src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+    * @return
+    */
+  private def deltaWithBlacklist(blacklist: List[BlacklistedTerm], delta: Delta = Delta()) = {
+    delta.copy(
+      blacklist = Some(blacklist)
+    )
+  }
+
+  private class MessageHandler extends IMessageHandler {
+    override def notifyException(exception: Throwable, phase: ExceptionPhase): Unit = {
+      logError("Service Bus client threw error while processing message.", exception)
+    }
+
+    /**
+      * Called by the QueueClient thread when a message arrives on the Service Bus.
+      *
+      * Note: this is configured in [[TransformManager.startQueueClient]] such that it's called serially (only 1 thread
+      *       will execute it at a time) which is necessary for the concurrency correctness of this module.
+      */
+    override def onMessageAsync(message: IMessage): CompletableFuture[Void] = {
+      // Wait for the previous update of transform context to settle
+      // to ensure we're calculating the delta against the latest context
+      writeLock.lock()
+      writeLock.unlock()
+
+      // Read the service bus message and build delta using data store.
+      val delta = Option(message.getProperties.getOrDefault("dirty", null)) match {
+        case Some(value) => value match {
+          case "settings" =>
+            val siteSettings = configManager.fetchSiteSettings()
+            deltaWithSettings(siteSettings)
+          case "watchlist" =>
+            val langToWatchlist = configManager.fetchWatchlist()
+            deltaWithWatchlist(langToWatchlist)
+          case "blacklist" =>
+            val blacklist = configManager.fetchBlacklist()
+            deltaWithBlacklist(blacklist)
+          case unknown =>
+            logError(s"Service Bus client received unexpected update request. Ignoring.: $unknown")
+            Delta()
+        }
+        case None =>
+          logError(s"Service Bus client received unexpected message. Ignoring.: ${message.toString}")
+          Delta()
+      }
+
+      // Block for up to two minutes for a Spark thread to acknowledge the updated
+      // state. If we time out, assume that this TransformManager instance has been
+      // replaced (Spark context restarted & checkpoint was discarded), and shut down to
+      // allow our successor to handle the message instead.
+      if (!shouldUpdate.offer(delta, 2, TimeUnit.MINUTES)) {
+        logDebug("Shutting down Service Bus client: timeout exceeded.")
+
+        // Shut down client
+        queueClient.closeAsync()
+
+        throw new Exception("No Spark thread acknowledged the update message within the timeout.")
+      }
+
+      CompletableFuture.completedFuture(null)
+    }
+  }
+}
+
+private object TransformManager {
+
+  /**
+    * Holds the next set of values for the fields of the current transform context. If the value of a field here is
+    * empty, then the corresponding field of the transform context during update will be left as is.
```

Comment:
I wonder if this can be made more explicit (and therewith the comment redundant_ by having a top-level constant called `NoChange` that's an alias for `None`.